hwsecurity/fido2/src/main/java/de/cotech/hw/fido2/internal/ctap2/commands/getAssertion/AuthenticatorGetAssertionResponseFactory.java

```diff
@@ -76,13 +80,27 @@ private AuthenticatorGetAssertionResponse readAuthenticatorGetAssertionResponse(
         PublicKeyCredentialUserEntity publicKeyCredentialUserEntity =
                 readPublicKeyCredentialUserEntity(user);
 
+        byte[] decryptedHmacSecretValue = null;
+        AuthenticatorData decodedAuthData = authenticatorDataParser.fromBytes(authData.getBytes());
+        if (decodedAuthData.hasExtensionData()) {
+            DataItem hmacSecretValue = decodedAuthData.extensions().get(HMAC_SECRET_EXTENSION_ID);
+            if (hmacSecretValue != null) {
+                if (hmacSecretValue.getMajorType() != MajorType.BYTE_STRING) {
+                    throw new IOException("hmac-secret response is not a byte string");
+                }
+                byte[] hmacSecretBytes = ((ByteString) hmacSecretValue).getBytes();
+                decryptedHmacSecretValue = authenticatorGetAssertion.pinToken().decrypt(hmacSecretBytes);
```

Comment:
Return side of the hmac-secret getAssertion flow, and the reason why I did all this wiring of the PinToken: the hmac-secret response value must be decrypted using the platform<->authenticator shared secret in order to be useful.

That's done here in the response factory.